Client/Frontend/Widgets/PhotonActionSheetProtocol.swift

```diff
@@ -162,23 +162,20 @@ extension PhotonActionSheetProtocol {
         }
         
         let pinToTopSites = PhotonActionSheetItem(title: Strings.PinTopsiteActionTitle, iconString: "action_pin") { action in
-            guard let url = tab.url?.displayURL else { return }
+            guard let url = tab.url?.displayURL,
+                  let sql = self.profile.history as? SQLiteHistory else { return }
             let absoluteString = url.absoluteString
             
-            let sql = self.profile.favicons as! SQLiteHistory
-            sql.getSitesForURLs([absoluteString]).uponQueue(.main) { result in
-                guard let results = result.successValue else {
+            sql.getSitesForURLs([absoluteString]) >>== { result in
+                guard let site = result.first(where: { $0 != nil }) else {
```

Comment:
ah. I see whats going on. Result is a `cursor` thats why this looks so funky. This should be `result.asArray()?.first`. Sorry about that! didnt realize what was going on.

Reply:
This results in `Cannot use optional chaining on non-optional value of type '[Site?]'`